### PR TITLE
Java optional example

### PR DESCRIPTION
**Pull request by cmiller-wikia**
I was asked to provide an example of how to convert code using Optional#isPresent to its type-safe alternative.

### CODE REVIEW

File: Java/Optional_example.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+              .build();
+     });
+```
+
+This could be made clearer by extracting the boiler-plate into a helper function.
+
+```java
+   return this.userPreferencesService.lookup(userId)
+     .map(userPreferences -> Response.ok(userPreferences).build())
+   	 .orElseGet(() -> StandardErrors.notFound(
```
Comment (drsnyder) [inline Java/Optional_example.md]: Is [`.orElseThrow`](https://github.com/Wikia/pandora/commit/55b7ccc#diff-950d57045a5076ec8d468966a29f99e2R56) ok here as well using a custom [Jersey exception](https://jersey.java.net/documentation/latest/representations.html#d0e6559)?

**Comment by drsnyder** [comment]
:+1: thanks for cranking that out.

**Comment by lizlux** [comment]
ping @cmiller-wikia - just noticed this old PR. Should it be merged?

**Comment by TK-999** [comment]
Closing as an old PR.

**Pull request closed without merging**